src/runtime/action/index.ts

```diff
@@ -17,26 +17,35 @@
  *
  * Docs: https://svelte.dev/docs#template-syntax-element-directives-use-action
  */
-export interface ActionReturn<Parameter = any> {
-	update?: (parameter: Parameter) => void;
-	destroy?: () => void;
-}
+export type ActionReturn<Parameter = any> = void extends Parameter
+	? {
+		destroy?: () => void;
+	}
+	: {
+		update?: (parameter: Parameter) => void;
+		destroy?: () => void;
+	}
```

Comment:
or to avoid breaking change, we can do this instead

```suggestion
interface ActionReturn<Parameter = void> {
	update?: void extends Parameter ? void : (parameter: Parameter) => void;
	destroy?: () => void;
}
```

Reply:
This is a nice middle ground for me 👍